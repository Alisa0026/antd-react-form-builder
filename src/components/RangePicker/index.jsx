import React from 'react'
import _ from 'lodash'
import moment from 'moment'
import BasicItem from '../BasicItem'
import AntdDatePicker from 'antd/lib/date-picker'
import FormItemComponentDecorator from '../../decorator/FormItemComponent'

const { MonthPicker, RangePicker } = AntdDatePicker;

function component(BasicItemComponent){
  @FormItemComponentDecorator
  class FRangePicker extends React.Component {

    render(){
      let { ...other } = this.props;
      other.targetComponent = RangePicker;
      other.type = "rangepicker";
      if(_.isArray(other.value)){
        other.value.forEach((v,k)=>{
          if(_.isString(v)){
            other.value[k] = new moment(v,other.format || "YYYY-MM-DD")
          }
        })
      }
      this.propsAdapter(other);
      return (
        <BasicItemComponent { ...other }/>
      )
    }
    
  }
  return FRangePicker;
}

export default component(BasicItem) 




