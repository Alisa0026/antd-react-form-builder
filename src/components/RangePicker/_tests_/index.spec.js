import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import moment from 'moment'
import LocaleProvider from 'antd/lib/locale-provider'
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import RangePicker from '../index'

describe('RangePicker',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <RangePicker 
        required
        className="test"
        label="name"
        name="test"
        value={
          ["2016-04-05 00:00:00","2016-04-05 00:00:00"]
        }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(RangePicker); 
  it("stirng value should be rendered correctly",function(){
    const wrapper = mount(
      <RangePicker 
        required
        className="test"
        label="name"
        name="test"
        value={
          ["2016-04-05","2016-04-05"]
        }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(antdInputMount.props().value).toEqual(
      [new moment("2016-04-05","YYYY-MM-DD"),new moment("2016-04-05","YYYY-MM-DD")]
    );
  })
  it("moment value should be rendered correctly",function(){
    var value = [new moment("2016-04-05","YYYY-MM-DD"),new moment("2016-04-05","YYYY-MM-DD")]
    const wrapper = mount(
      <RangePicker 
        required
        className="test"
        label="name"
        name="test"
        value={ value }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(antdInputMount.props().value).toEqual(value);
  })
  it("should have hidden <Input /> ",function(){
    var value = [new moment("2016-04-05","YYYY-MM-DD"),new moment("2016-04-05","YYYY-MM-DD")]
    const wrapper = mount(
      <RangePicker 
        required
        className="test"
        label="name"
        name="test"
        value={ value }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    expect(wrapper.childAt(1).childAt(0).props().value).toEqual(Math.floor(+value[0] / 1000));
    expect(wrapper.childAt(2).childAt(0).props().value).toEqual(Math.floor(+value[1] / 1000));
    //expect(wrapper.childAt(1).props().name).toEqual('test');
  })
})

