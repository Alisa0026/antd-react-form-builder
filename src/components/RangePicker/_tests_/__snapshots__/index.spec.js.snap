// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangePicker renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control has-success"
      >
        <span
          class="ant-calendar-picker"
        >
          <span
            class="ant-calendar-picker-input ant-input"
          >
            <input
              class="ant-calendar-range-picker-input"
              placeholder="开始日期"
              readonly=""
              value="2016-04-05"
            />
            <span
              class="ant-calendar-range-picker-separator"
            >
               ~ 
            </span>
            <input
              class="ant-calendar-range-picker-input"
              placeholder="结束日期"
              readonly=""
              value="2016-04-05"
            />
            <i
              class="anticon anticon-cross-circle ant-calendar-picker-clear"
            />
            <span
              class="ant-calendar-picker-icon"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <span>
    <input
      class="ant-input"
      name="test"
      type="hidden"
      value="1459785600"
    />
  </span>
  <span>
    <input
      class="ant-input"
      name="test"
      type="hidden"
      value="1459785600"
    />
  </span>
</span>
`;
